trio_run_in_process/run_in_process.py

```diff
@@ -223,3 +224,65 @@ async def open_worker_process() -> AsyncIterator[WorkerProcessAPI]:
                 yield worker
     finally:
         worker._dead = True
+
+
+class WorkerPool(WorkerPoolAPI):
+    def __init__(self, max_workers: int) -> None:
+        self._max_workers = max_workers
+        self._num_workers = 0
+        self._send_channel, self._receive_channel = trio.open_memory_channel[
+            WorkerProcessAPI
+        ](max_workers)
+        self._exit_stack = AsyncExitStack()
+        self._is_open = False
+
+    @asynccontextmanager
+    async def _reserve_worker(self) -> AsyncIterator[WorkerProcessAPI]:
+        try:
+            worker = self._receive_channel.receive_nowait()
+        except trio.WouldBlock:
+            if self._num_workers < self._max_workers:
+                self._num_workers += 1
+                logger.debug("Created new worker")
```

Comment:
Might be nice to have some sort of identifier for the pool.  Also, maybe to move this to after the creation so that we could log the PID?

Reply:
Yeah, that'd be nice. I've added an ID arg to WorkerPool's constructor, using an uuid4 if none is passed in